### PR TITLE
chore: add E2E tests for sidecars

### LINKED ISSUE #576: E2E tests for sidecars
While working on `celestia-node` e2e tests, found out there are some limitations in using sidecars. For example, it failed to read a file from an obsy sidecar. Not sure if there was an issue with `knuu` or with that particular sidecar or its configs. Therefore, having a comprehensive e2e tests that tests the required functions on a sidecar seems necessary. The tests can include, but not limited to:  Execute a command in a sidecar, read a file from a running sidecar, ... 

Currently there is only reading logs from a sidecar and of course tests for the knuu modules, which are developed as sidecars. e,g: NetShaper, TShark,...

### PR DESCRIPTION
**Pull request by mojtaba-esk**
Closes #576 

## Summary by CodeRabbit

- **New Features**
	- Introduced new test functions for validating command execution and file downloading within sidecar instances.
	- Added logging functionality tests for sidecar containers.
	- Established a new test suite for comprehensive end-to-end testing of the sidecars component.

- **Bug Fixes**
	- Removed outdated sidecar logging tests to streamline the testing process.

- **Documentation**
	- Enhanced error handling and validation across new test functions to improve reliability.

### CODE REVIEW

File: e2e/sidecars/cmd_exec_test.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+	s.Require().NoError(target.Build().SetImage(ctx, alpineImage))
+	s.Require().NoError(target.Build().SetArgs("tail", "-f", "/dev/null")) // Keep the container running
+	s.Require().NoError(target.Build().Commit(ctx))
+
+	sidecar := &testSidecar{
+		StartCommand: []string{"sh", "-c", "sleep infinity"},
+	}
+
+	s.Require().NoError(target.Sidecars().Add(ctx, sidecar))
+	s.Require().NoError(target.Execution().Start(ctx))
```
Comment (MSevey) [inline e2e/sidecars/cmd_exec_test.go:28]: This looks like duplicate code in all the tests that could be refactored into a single helper function.

File: e2e/sidecars/logs_test.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+	}
+
+	s.Require().NoError(target.Build().SetImage(ctx, alpineImage))
+	s.Require().NoError(target.Build().SetStartCommand("sh", "-c", "sleep infinity"))
+	s.Require().NoError(target.Build().Commit(ctx))
+	s.Require().NoError(target.Sidecars().Add(ctx, sidecar))
+	s.Require().NoError(target.Execution().Start(ctx))
+
+	// Wait for a short duration to allow log generation
+	time.Sleep(5 * time.Second)
```
[the remark this answers is not in the document]
Comment (MSevey) [reply e2e/sidecars/logs_test.go:34]: Can we use the retry mechanism helper here? 
https://github.com/celestiaorg/knuu/blob/f86e86ca65c758ceaef4cf72d78d1352a2756089/e2e/suite.go#L85

**Review by MSevey — CHANGES_REQUESTED** [review]
Looks good, I think we should refactor the common test code to clean things up and use the Retry helper instead of the sleep.

**Review by smuu — APPROVED** [review]

File: e2e/sidecars/cmd_exec_test.go [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+	// Create a file in the sidecar instance
+	out, err := sidecar.Instance().Execution().ExecuteCommand(ctx, command)
+	s.Require().NoError(err, "executing command output: %v", out)
+
+	outTrimmed := strings.TrimSpace(out)
+	s.Assert().Equal(cmdMsg, outTrimmed)
+}
```
[the remark this answers is not in the document]
Comment (mojtaba-esk) [reply e2e/sidecars/cmd_exec_test.go:36]: Thanks for the comment. However, those you mentioned are out of the scope of this test. This test focuses on sidecars only and not to test the command execution function.

File: e2e/sidecars/file_test.go [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+	// Create a file in the sidecar instance
+	out, err := sidecar.Instance().Execution().ExecuteCommand(ctx, "echo", "-n", fileContent, ">", filePath)
+	s.Require().NoError(err, "executing command output: %v", out)
+
+	gotContent, err := sidecar.Instance().Storage().GetFileBytes(ctx, filePath)
+	s.Require().NoError(err, "Error getting file bytes")
+
+	s.Assert().Equal(fileContent, string(gotContent))
+}
```
[the remark this answers is not in the document]
Comment (mojtaba-esk) [reply e2e/sidecars/file_test.go:37]: thanks I applied it and will push it asap

File: e2e/sidecars/logs_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (mojtaba-esk) [reply e2e/sidecars/logs_test.go:34]: Thanks `s.Require().Eventually` seems to be a neat option. will apply it

File: e2e/sidecars/logs_test.go [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+	logStream, err := sidecar.Instance().Monitoring().Logs(ctx)
+	s.Require().NoError(err)
+	defer logStream.Close()
+
+	logs, err := io.ReadAll(logStream)
+	s.Require().NoError(err)
+
+	logOutput := string(logs)
+	s.Contains(logOutput, expectedLogMsg)
+}
```
[the remark this answers is not in the document]
Comment (mojtaba-esk) [reply e2e/sidecars/logs_test.go:45]: The test suit itself has a cleanup mechanism. so no need to add it here

File: e2e/sidecars/cmd_exec_test.go (hunk 1 above)
Comment (MSevey) [reply e2e/sidecars/cmd_exec_test.go:28]: No, all of this code seems duplicate
```
	target, err := s.Knuu.NewInstance(namePrefix + "-target")
	s.Require().NoError(err)

	ctx := context.Background()
	s.Require().NoError(target.Build().SetImage(ctx, alpineImage))
	s.Require().NoError(target.Build().SetArgs("tail", "-f", "/dev/null")) // Keep the container running
	s.Require().NoError(target.Build().Commit(ctx))

	sidecar := &testSidecar{
		StartCommand: []string{"sh", "-c", "sleep infinity"},
	}

	s.Require().NoError(target.Sidecars().Add(ctx, sidecar))
```

Comment (MSevey) [reply e2e/sidecars/cmd_exec_test.go:28]: the start command can be an input to the function. 
all the NewInstance, and commands on `target` seem like they are just copy paste across all these tests.

Comment (mojtaba-esk) [reply e2e/sidecars/cmd_exec_test.go:28]: done

File: e2e/sidecars/logs_test.go [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+	}
+
+	s.Require().NoError(target.Build().SetImage(ctx, alpineImage))
+	s.Require().NoError(target.Build().SetStartCommand("sh", "-c", "sleep infinity"))
+	s.Require().NoError(target.Build().Commit(ctx))
+	s.Require().NoError(target.Sidecars().Add(ctx, sidecar))
+	s.Require().NoError(target.Execution().Start(ctx))
+
+	// Wait for a short duration to allow log generation
+	s.Require().Eventually(func() bool {
```
[the remark this answers is not in the document]
Comment (mojtaba-esk) [reply e2e/sidecars/logs_test.go:38]: me neither, yeah it is cool

**Review by MSevey — APPROVED** [review]

**Pull request merged**